src/input/styled-components.js

```diff
@@ -232,9 +232,15 @@ function getInputContainerColors(
   };
 }
 
-export const getInputContainerStyles = (
-  props: SharedPropsT & {$theme: ThemeT},
-) => {
+export const getInputContainerStyles = (props: {
+  $adjoined: $Keys<typeof ADJOINED>,
```

Comment:
Why duplicating the props here and not reuse the exported type?

Reply:
the fields here are a subset of what is provided in another component. because we get type checking now.. the other component fails because a field expected in `SharedPropsT` is not provided